flake.nix

```diff
@@ -58,9 +58,26 @@
 
           shellHook = ''
             echo "Setting up default dev shell..."
+            DEPS_CACHE_TMP="$(mktemp -dt gensql.query.clj.cache.XXXX)"
+            echo "Copying readonly clojure cache to read-write location at:"
+            echo "$DEPS_CACHE_TMP"
+            rsync -Laz ${depsCache}/ $DEPS_CACHE_TMP/
```

Comment:
OK, I checked out your branch, ran `direnv allow`, and got a trillion "Permission denied" errors:

```shell
Code/probcomp/GenSQL.query [ HEAD](b787bdc) [📦 v1.0.0][ v21.0.1][ v20.11.1]❯
direnv allow
direnv: loading ~/Code/probcomp/GenSQL.query/.envrc
direnv: using flake
evaluating derivation 'git+file:///Users/matthew/Code/probcomp/GenSQL.query#devShells.x86_64-darwin.default'direnv: ([/run/current-system/sw/bin/direnv export zsh]) is taking a while to execute. Use CTRL-C to give up.
direnv: nix-direnv: renewed cache
Setting up default dev shell...
Copying readonly clojure cache to read-write location at:
/var/folders/kn/88hthhl94370gvp0f8hsz1w00000gn/T/nix-shell.dilDao/gensql.query.clj.cache.OZla
rm: cannot remove '/var/folders/kn/88hthhl94370gvp0f8hsz1w00000gn/T/nix-shell.dilDao/gensql.query.clj.cache.OZla/.clojure/tools/tools.edn': Permission denied
rm: cannot remove '/var/folders/kn/88hthhl94370gvp0f8hsz1w00000gn/T/nix-shell.dilDao/gensql.query.clj.cache.OZla/.clojure/deps.edn': Permission denied
rm: cannot remove '/var/folders/kn/88hthhl94370gvp0f8hsz1w00000gn/T/nix-shell.dilDao/gensql.query.clj.cache.OZla/.gitlibs/_repos/https/github.com/cognitect-labs/test-runner/config': Permission denied
rm: cannot remove '/var/folders/kn/88hthhl94370gvp0f8hsz1w00000gn/T/nix-shell.dilDao/gensql.query.clj.cache.OZla/.gitlibs/_repos/https/github.com/clojure/tools.build/config': Permission denied
rm: cannot remove '/var/folders/kn/88hthhl94370gvp0f8hsz1w00000gn/T/nix-shell.dilDao/gensql.query.clj.cache.OZla/.gitlibs/libs/io.github.clojure/tools.build/8e78bccc35116f6b6fc0bf0c125dba8b8db8da6b/LICENSE': Permission denied
rm: cannot remove '/var/folders/kn/88hthhl94370gvp0f8hsz1w00000gn/T/nix-shell.dilDao/gensql.query.clj.cache.OZla/.gitlibs/libs/io.github.clojure/tools.build/8e78bccc35116f6b6fc0bf0c125dba8b8db8da6b/test-data/p3/deps.edn': Permission denied
...
```

I'm not 100% certain which strategy rsync is using, but it apparently needs writable files, which we want too, I assume. Since rsync's `-a` option preserves the original read-only permissions, I expanded out the `-a`, removed the `-p`, and added a `chmod` option to make sure things are user/group-writable:

`rsync -LzrltgoD --chmod=ug+w ${depsCache}/ $DEPS_CACHE_TMP/`

(Side note: is using compression with the `-z` option worth it? I would think that would defeat DMA by forcing to data down to the CPU, to be compressed, uncompressed, and finally written back out.)

Reply:
There's another issue, which is that the writable cache isn't shared. Every time a shell enters/starts in the query dir, a _new_ temp cache is created. This could be a bit confusing, if updates in one shell are invisible to others. And with direnv, just cd-ing out and back makes a new one.

Reply:
Hmm... i'm thinking thru the consequences of not sharing the deps cache between shells. It would mean when you open new shells you have to redownload, right? I do see that being a bit annoying/confusing. To avoid that we would need an explicit step to reset to the locked deps. Would that be preferable to you?

we could for example create a temporary directory in $PWD with a fixed name.

Reply:
And you're right about compression, i timed it and consistently takes 0.5s instead of 1.5-2s when you remove compression.